### PR TITLE
fix: Use `options.reportAccessibilityIdentifier` for Breadcrumbs and UIEvents

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

We added the new options `reportAccessibilityIdentifier` but did not use it in other places.
accessibilityIdentifier may contain PII also in this other places.

## :green_heart: How did you test it?

Unit tests

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] I updated the wizard if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ### Fixes
 
 - Session replay touch tracking race condition (#4548)
+- Use `options.reportAccessibilityIdentifier` for Breadcrumbs and UIEvents ()
```
Comment (brustolin) [inline CHANGELOG.md:8]:
```suggestion
- Use `options.reportAccessibilityIdentifier` for Breadcrumbs and UIEvents (#4569) 
```

File: Sources/Sentry/SentryBreadcrumbTracker.m [hunk 2]
```diff
@@ -43,6 +45,20 @@ - (void)dealloc
 }
 #endif // !TARGET_OS_WATCH
 
+// Convenient initializer for tests
+- (instancetype)init
+{
+    return [self initReportAccessibilityIdentifier:YES];
+}
```
Comment (philipphofmann) [inline Sources/Sentry/SentryBreadcrumbTracker.m:52]: `m`: This opens a door for initializing the `SentryBreadcrumbTracker` and setting the `_reportAccessibilityIdentifier` to `YES` by mistake. I prefer removing this initializer.

File: Tests/SentryTests/Integrations/Breadcrumbs/SentryBreadcrumbTrackerTests.swift [hunk 3]
```diff
@@ -230,6 +230,82 @@ class SentryBreadcrumbTrackerTests: XCTestCase {
...
+        let tracker = SentryBreadcrumbTracker(reportAccessibilityIdentifier: false)
+        tracker.start(with: delegate)
+        tracker.startSwizzle()
+        
+        let button = UIButton()
+        button.accessibilityIdentifier = "TestAccessibilityIdentifier"
+        
+        swizzlingWrapper.execute(action: "methodPressed:", target: self, sender: button, event: TestEvent(touchedView: button))
+        
+        guard let crumb = delegate.addCrumbInvocations.invocations.first(where: { $0.category == "touch" }) else {
```
Comment (philipphofmann) [inline Tests/SentryTests/Integrations/Breadcrumbs/SentryBreadcrumbTrackerTests.swift:251]: `l`: We always use XCTUnwrap in the tests. I would prefer sticking to this approach, and the syntax is shorter.

File: Tests/SentryTests/Integrations/Breadcrumbs/SentryBreadcrumbTrackerTests.swift [hunk 4]
```diff
@@ -230,6 +230,82 @@ class SentryBreadcrumbTrackerTests: XCTestCase {
         XCTAssertEqual(payload["message"] as? String, "methodPressed:")
     }
     
+    func testTouchBreadcrumb_DontReportAccessibilityIdentifier() throws {
```
Comment (philipphofmann) [inline Tests/SentryTests/Integrations/Breadcrumbs/SentryBreadcrumbTrackerTests.swift:233]: Neat tests 👍

**Review by philipphofmann — APPROVED** [review]
LGTM, with minor suggestions.

File: Tests/SentryTests/Integrations/Breadcrumbs/SentryBreadcrumbTrackerTests.swift (hunk 3 above)
Comment (brustolin) [reply Tests/SentryTests/Integrations/Breadcrumbs/SentryBreadcrumbTrackerTests.swift:251]: Ctrl+C Ctrl+V 😅
I just changed the important parts.

**Pull request merged**